src/files/models/Dijkstra.java

```diff
@@ -131,24 +94,34 @@ public Vertex dijkstra(Vertex startNode, Vertex endNode) {
         Vertex current;
 
         while (graphTreeSet.size() != 0) {
-            System.out.println(graphTreeSet.first().name+graphTreeSet.first().distance);
             current = graphTreeSet.first();
             System.out.println("  CURRENT:  " + current.name);
+
             for (Edge edge : current.edges) {
+
                 if (current.distance != infinity && edge.distance + current.distance < edge.getToVertex().distance) {
                     edge.getToVertex().setDistance(edge.distance + current.distance);
                     edge.getToVertex().setPredecessor(current);
                     graphTreeSet.remove(edge.getToVertex());
                     graphTreeSet.add(edge.getToVertex());
                 }
+
                 System.out.println("Evaluating:  " + edge.getFromVertex().name + "  ->  " + edge.getToVertex().name
                         + "  dist:  " + edge.getToVertex().distance);
             }
-            //graphTreeSet.forEach((vertex -> System.out.print(vertex.name + ": " + vertex.distance + "   |")));
+
             graphTreeSet.remove(current);
             System.out.println("removed:  " +current.name);
         }
 
-        return endNode;
+        Vertex resultCurrent = endNode;
```

Comment:
not sure i understand what resultCurrent is used for

Reply:
changed it so it returns an arraylist